lightning-background-processor/src/lib.rs

```diff
@@ -218,16 +218,24 @@ impl BackgroundProcessor {
 			let mut last_prune_call = Instant::now();
 			let mut have_pruned = false;
 
+			// When considering how long its taken since the last timer tick, we don't want to
+			// count any time spent in user code, especially since event processing can block on
+			// disk writes. Thus, we track how long we spent in event handling here.
+			let mut ev_handle_time_since_last_ping = Duration::from_millis(0);
 			loop {
-				peer_manager.process_events();
+				let ev_handle_start = Instant::now();
+				peer_manager.process_events(); // Note that this may block on ChannelManager's locking
 				channel_manager.process_pending_events(&event_handler);
 				chain_monitor.process_pending_events(&event_handler);
+				ev_handle_time_since_last_ping += ev_handle_start.elapsed();
 				let updates_available =
 					channel_manager.await_persistable_update_timeout(Duration::from_millis(100));
 				if updates_available {
+					let persist_start = Instant::now();
```

Comment:
Considering the 100ms timeout, may be simpler just to use one timer ending outside the `if` block.

Reply:
I'm not sure I understand, are you saying just move this timer outside the if block?

Reply:
Meant we could potentially combine `ev_handle_start` and `persist_start` timers into a single timer.

Reply:
We don't know how long `await_persistable_update_timeout` takes, though, and the goal here is (mostly) to measure how long it took as an indirect way to figure out whether we went to background on, eg, iOS.

Reply:
Hmm... isn't that what the timeout is for? Maybe I'm misunderstanding how it works.

Reply:
I dropped all the addition stuff, it was actually incorrect cause of lack of rebase anyway, does the comment in the if block make sense?